unit_tests/test_tlslite_messages.py

```diff
@@ -2821,6 +2940,38 @@ def test_parse_with_cert(self):
         self.assertEqual(cert.certChain.x509List[0].writeBytes(),
                 self.der_cert)
 
+    def test_parse_tls1_3_with_cert(self):
+        cert = Certificate(CertificateType.x509, (3, 4))
+        self.assertEqual(0x0001fa, len(self.der_cert))
+        parser = Parser(
+                bytearray(#b'\x0b' +  # type of message - certificate
+                          b'\x00\x02\x09' +  # length of handshake message
+                          b'\x00' +  # length of certificate request context
+                          b'\x00\x02\x05' +  # length of certificate list
+                          b'\x00\x01\xfa' +  # length of certificate
+                          self.der_cert +
+                          b'\x00\x06' +  # length of extensions
+                          b'\x00\xff' +  # type of first extension
+                          b'\x00\x02' +  # length of first extension
+                          b'\xde\xad'))
+
+        cert = cert.parse(parser)
+
+        self.assertIsNotNone(cert)
+        self.assertEqual(cert.certificate_request_context, bytearray(b''))
+        self.assertIsNotNone(cert.certificate_list)
+        self.assertEqual(len(cert.certificate_list), 1)
+        self.assertIsInstance(cert.certificate_list[0], CertificateEntry)
+        entry = cert.certificate_list[0]
+        self.assertEqual(entry.certificate.writeBytes(), self.der_cert)
+        self.assertEqual(len(entry.extensions), 1)
+        self.assertEqual(entry.extensions[0].extData, bytearray(b'\xde\xad'))
+        self.assertIsNotNone(cert.certChain)
+        self.assertIsInstance(cert.certChain, X509CertChain)
+        self.assertEqual(len(cert.certChain.x509List), 1)
+        self.assertEqual(cert.certChain.x509List[0].writeBytes(),
+                self.der_cert)
+
     def test_parse_with_openpgp_type(self):
         cert = Certificate(CertificateType.openpgp)
```

Comment:
I don't see much benefit in handling with the same class the TLS1.2 and TLS1.3 formatted structures, nevertheless, r+